ganttproject/src/main/java/net/sourceforge/ganttproject/storage/SqlProjectDatabaseImpl.kt

```diff
@@ -326,11 +356,31 @@ internal class SqlTaskCustomPropertiesUpdateBuilder(
       .and(TASKCUSTOMCOLUMN.COLUMN_ID.notIn(customProperties.customProperties.map { it.definition.id }))
       .getSQL(ParamType.INLINED)
 
+  private fun generateDeleteDto(customProperties: CustomPropertyHolder): OperationDto.DeleteOperationDto =
+    OperationDto.DeleteOperationDto(
+      TASKCUSTOMCOLUMN.name.lowercase(),
+      listOf(
+        Triple(TASKCUSTOMCOLUMN.UID.name, BinaryPred.EQ, task.uid),
+      ),
+      listOf(
+        Triple(TASKCUSTOMCOLUMN.COLUMN_ID.name, RangePred.NOT_IN, customProperties.customProperties.map { it.definition.id } )
+      )
+    )
+
   private fun generateDeleteStatementAllColumns(dsl: DSLContext): String =
     dsl.deleteFrom(TASKCUSTOMCOLUMN)
       .where(TASKCUSTOMCOLUMN.UID.eq(task.uid))
       .getSQL(ParamType.INLINED)
 
+  private fun generateDeleteDtoAllColumns(): OperationDto.DeleteOperationDto =
+    OperationDto.DeleteOperationDto(
+      TASKCUSTOMCOLUMN.name.lowercase(),
+      listOf(
+        Triple(TASKCUSTOMCOLUMN.UID.name, BinaryPred.EQ, task.uid),
+      ),
+      listOf()
```

Comment:
any sense to add listOf() as the default value of this argument?

Reply:
For delete -- yup, done